### DISCUSSION TITLE
Identity Federation - Authenticating with AD Service Principals

### DISCUSSION THREAD

**Discussion by kinihun** [keycloak/keycloak#12555] [Q&A]
Hi,
I am using Apicurio (a schema registry), Keycloak as my Identity Broker and I've setup Active Directory (AD) as my Identity Provider (IDP). My users can login to Apicurio UI using SSO however I would like non-users (other applications) to read and write to the Apicurio API (not UI).
How do I setup other applications to use their AD-issued identities (Service Principal credentials) to authenticate via Keycloak?

**Comment by VidhiPatel** [comment]
I am trying to achieve something similar, was there any solution for this?
